index.bs

```diff
@@ -488,13 +490,238 @@ partial interface NeuralNetworkContext {
+        - *returnSequence*: an optional {{boolean}} indicating whether to also return the entire sequence with every cell outputs from each time step in it in addition to the cell output of the last time step. Default to false.
+        - *direction*: an optional {{RecurrentNetworkDirection}}. The processing direction of the input sequence. When set to *"both"*, the size of the first dimension of the weight and the bias tensor shapes must be 2, and the input is processed in both directions.
+        - *layout*: an optional {{RecurrentNetworkWeightLayout}}. The ordering of the weight and bias vectors for the internal gates of GRU, specifically the *update (z)*, *reset (r)*, and *new (n)* gate, as indicated in the second dimension of the weight and bias tensor shape. When not specified, the default layout is *"zrn"*.
+        - *activations*: an optional sequence of {{RecurrentNetworkActivation}}. A pair of activation functions with the first function used for the update and reset gate, and the second used for the new gate. When not specified, it's default to the sigmoid ("sigmoid") and the hyperbolic tangent ("tanh") function respectively.
+
+    **Returns:** a sequence of {{Operand}}. The first element of the sequence is a 3-D tensor of shape [num_directions, batch_size, hidden_size], the cell output from the last time step of the network. Additionally, if *returnSequence* is set to true, the second element is the 4-D output tensor of shape [steps, num_directions, batch_size, hidden_size] containing every cell outputs from each time step in the temporal sequence.
+
+    <div class="note">
+    The behavior of this operation with default argument values can be generically emulated from the usage of other operations as follow. However, user agents typically have a more efficient implementation for it, therefore its usage is encouraged from the performance standpoint.
+    <pre highlight="js">
+    let hiddenState = initialHiddenState;
+    let slots = (direction == "both" ? 2 : 1);
+    let sequence = null;
+
+    let cellWeight = [];
+    let cellRecurrentWeight = [];
+    let cellBias = [];
+    let cellRecurrentBias = [];
+
+    for (let slot = 0; slot < slots; ++slot) {
+      cellWeight.push(nn.squeeze(nn.slice(weight, [slot, 0, 0], [slot + 1, -1, -1])), [0]);
+      cellRecurrentWeight.push(nn.squeeze(nn.slice(recurrentWeight, [slot, 0, 0], [slot + 1, -1, -1])), [0]);
+      cellBias.push(bias ? (nn.squeeze(nn.slice(bias, [slot, 0], [slot + 1, -1]), [0])) : null);
+      cellRecurrentBias.push(recurrentBias ? (nn.squeeze(nn.slice(recurrentBias, [slot, 0], [slot + 1, -1]), [0])) : null);
+    }
+
+    for (let step = 0; step < steps; ++step) {
+      let cellHidden = [];
+      let cellOutput = null;
+
+      for (let slot = 0; slot < slots; ++slot) {
+        cellHidden.push(nn.squeeze(nn.slice(hidden, [slot, 0, 0], [slot + 1, -1, -1]), [0]));
```

Comment:
`hidden` is not defined. Is it `hiddenState`?

Reply:
Super. I did miss renaming that one.

Reply:
Fixed.